compiler/src/iree/compiler/Codegen/ExternalInterfaces/GPUEncodingExternalModels.cpp

```diff
@@ -48,6 +50,15 @@ using Codegen::TileMxNxK;
 
 namespace {
 
+static IREE::GPU::TargetAttr getGPUTargetNamedAttr(DictionaryAttr dictAttr) {
+  std::optional<NamedAttribute> targetAttr =
+      dictAttr.getNamed("iree.gpu.target");
+  if (!targetAttr) {
+    return {};
+  }
+  return cast<IREE::GPU::TargetAttr>(targetAttr->getValue());
+}
```

Comment:
Good point, I've made a similar change for CPU before! I think we can adapt the `getGPUTargetAttr` to take an attribute. If the incoming attribute is a `ExecutableTargetAttr`, we extract the gpu target from the configuration. If it is a `DictionaryAttr`, we extract the gpu target from the attribute. Otherwise, we return nullptr.

https://github.com/iree-org/iree/blob/c8ba691b42354957abc8130d750cf16cbf0ec0bf/compiler/src/iree/compiler/Codegen/Utils/GPUUtils.cpp#L970-L976


Here is an example:

https://github.com/iree-org/iree/blob/c8ba691b42354957abc8130d750cf16cbf0ec0bf/compiler/src/iree/compiler/Codegen/Utils/Utils.cpp#L67-L87

Reply:
Yeah, this makes sense! Thanks for the suggestion.